src/coreclr/tests/src/JIT/Directed/StructABI/structreturn.cs

```diff
@@ -1131,6 +1137,322 @@ public static void Test()
         ReturnVector4Wrapper();
         ReturnVector2x2Wrapper();
     }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static Vector<int> ReturnVectorInt()
+    {
+        return new Vector<int>();
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static Vector<float> ReturnVectorFloat()
+    {
+        return new Vector<float>();
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static Vector<T> ReturnVectorT<T>() where T : struct
+    {
+        return new Vector<T>();
+    }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    static Vector<T> ReturnVectorTWithMerge<T>(int v, T init1, T init2, T init3, T init4) where T : struct
+    {
+        if (v == 0)
```

Comment:
these `return` have `SIMD16` type, but when we create merge return LCL_VAR it gets `STRUCT` type, so when we assign `SIMD16` to `STRUCT` we are getting an assert.